### DISCUSSION TITLE
Solving the phase field model for brittle fracture using Multiapp and adptivity

### DISCUSSION THREAD

**Discussion by MartimSalgado** [idaholab/moose#18813] [Q&A Modules: Solid mechanics] [answered]
Dear Moosers,
I'm trying to solve the phase field model for brittle fracture using staggered (Picard) iterations. The history variable gets solved in the master (Tensor Mechanics) app and is transferred to the sub (phase field) app. The phase field variable is solved using a TransientApp and then transferred to the master (end of picard iteration). 
I'm getting incorrect crack propagation with angle cracks instead of horizontal cracks (as it should be). I think this may be due to the method used to transfer these variables or due to the meshes used.
I'm cloning the master mesh and independently using the same markers to h-adapt each mesh so I get the same mesh after each time step.
I'm using mesh function transfers (copy transfers are not possible since the transfer happens before the adaptivity step that would lead to the sub app having the same mesh as the master.

Any suggestions for fixing this problem? I'm sending both input files attached.

Thanks!
[sub_sent_staggered.zip](https://github.com/idaholab/moose/files/7154088/sub_sent_staggered.zip)

**Comment by jiangwen84** [comment, marked as answer]
First, have you tried the picard iteration without adaptivity? 

A few comments:
1. without adaptivity, you can use `MultiAppCopyTransfer`
2. I usually transfer `displacements` from master to sub and `damage` from sub to master. In both master and sub input files, I add materials to recompute all quantities, including fracture driving energy so that I do not need to transfer them.

**Reply by aeslaughter** [reply]
I created [LevelSetMeshRefinementTransfer](https://github.com/idaholab/moose/blob/next/modules/level_set/src/transfers/LevelSetMeshRefinementTransfer.C) 5 years ago for this same concept. There might be a better way to do it now, but if not you can take a look at it for some inspiration.

**Reply by MartimSalgado** [reply]
@jiangwen84 Thanks for your reply. I was trying to avoid recomputing the history variable, but I will follow your advice.
@aeslaughter that's the perfect method to couple the adaptivity of my apps, thanks!

**Comment by hugary1995** [comment]
There is an app for phase field fracture: [RACCOON](https://github.com/hugary1995/raccoon). This [tutorial](https://github.com/hugary1995/raccoon/tree/master/tutorials/adaptivity) is doing exactly what you are describing.

**Reply by MartimSalgado** [reply]
If only I had known.... Thanks @hugary1995! I'll definitely start using the RACOON
